crates/block-producer/src/transaction.rs

```diff
@@ -0,0 +1,131 @@
+use std::collections::BTreeSet;
+
+use miden_objects::{
+    accounts::AccountId,
+    notes::{NoteId, Nullifier},
+    transaction::{ProvenTransaction, TransactionId, TxAccountUpdate},
+    Digest,
+};
+
+use crate::{errors::VerifyTxError, store::TransactionInputs};
+
+/// A transaction whose proof has been verified.
+#[derive(Clone, PartialEq)]
+pub struct VerifiedTransaction(ProvenTransaction);
+
+/// A transaction who's proof has been verified, and which has been authenticated against the store.
+#[derive(Clone, PartialEq)]
+pub struct AuthenticatedTransaction {
+    inner: VerifiedTransaction,
+    store_account_state: Option<Digest>,
+    authenticated_notes: BTreeSet<NoteId>,
+    authentication_height: u32,
```

Comment:
Let's add doc comments to these fields - otherwise it is not immediately clear what they supposed to contain. For example, `authenticated_notes` is a set of notes that was unauthenticated in the transaction but is currently present in the store, right? (i.e., it is not a set of all authenticated notes in the transaction). Maybe it should be called `newly_authenticated_notes`?